### PR TITLE
feat: wallet history module core components

### PR DESCRIPTION
**Pull request by MosCD3**
# Summary of Changes

The history module will allow wallet users to track VC and general wallet events.
This PR is the base components needed to create the module. The PR will include the following 
- Initializing history module to record events on-demand
- Recording accepted credentials only for now
- Viewing history
- Setting history record options screen
This is considered the very basic components. Planned to be further developed after this is merged
- Adding queer service to record events before wallet initializes such as first creating PIN 
- Filtering and trimming old history items
- History details 
- Expanding history record events to be hooked to all VC related events

This is an overhaul for the previous PR that was created earlier before moving to credo
Major updates
- History is used as manager class initialized by app container instead of using provider
- Behind a feature flag toggled in Developer menu
- Hooked up to accepting new Crew. offer only as an MVP 

# Related Issues

Please reference here any issue #'s that are relevant to this PR, or simply enter "N/A" if this PR does not relate to any existing issues.

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [x] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [x] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [x] Updated documentation as needed for changed code and new or modified features;
- [x] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

**Comment by bryce-mcmath** [comment]
able to fix the `opposite operator` issue found by sonarcloud? The rest are mostly TODO warnings or not from your code, so not to worry about those, but I agree with the opposite operator suggestion

File: packages/legacy/core/App/modules/history/context/historyManager.tsx [hunk 1]
```diff
@@ -0,0 +1,245 @@
+/**
+ *  USAGE
+ * 
+        const { agent } = useAgent()
+
+        //Log history
+        const historyManager = container.resolve(TOKENS.FN_LOAD_HISTORY)(newAgent)
```
Comment (bryce-mcmath) [inline packages/legacy/core/App/modules/history/context/historyManager.tsx]: where is `newAgent` coming from in this usage example? Do you mean just `agent`?

Comment (MosCD3) [reply packages/legacy/core/App/modules/history/context/historyManager.tsx]: yes I meant "agent", fixed the typo

File: packages/legacy/app/ios/Podfile.lock
Comment (bryce-mcmath) [inline packages/legacy/app/ios/Podfile.lock:1]: There appears to be no changes to ios deps or npm deps in this PR so this file should not be changed as part of this PR

File: packages/legacy/app/ios/ariesbifold.xcodeproj/project.pbxproj
Comment (bryce-mcmath) [inline packages/legacy/app/ios/ariesbifold.xcodeproj/project.pbxproj:1]: This change should not be included, this diff is related to your local configuration

File: packages/legacy/core/App/modules/history/services/quee.service.tsx
Comment (bryce-mcmath) [inline packages/legacy/core/App/modules/history/services/quee.service.tsx:1]: Should this file be named `queue.service.tsx`?

**Review by bryce-mcmath — CHANGES_REQUESTED** [review]
Few more changes requested, sorry I didn't notice these the first go-round

File: packages/legacy/app/ios/Podfile.lock
Comment (MosCD3) [reply packages/legacy/app/ios/Podfile.lock:1]: done

File: packages/legacy/app/ios/ariesbifold.xcodeproj/project.pbxproj
Comment (MosCD3) [reply packages/legacy/app/ios/ariesbifold.xcodeproj/project.pbxproj:1]: done

File: packages/legacy/core/App/modules/history/services/quee.service.tsx
Comment (MosCD3) [reply packages/legacy/core/App/modules/history/services/quee.service.tsx:1]: done

**Comment by bryce-mcmath** [comment]
Looks good can we get it up to date with main?

**Comment by cvarjao** [comment]
@MosCD3 , just waiting for a rebase so we can merge this PR. Maybe give us a heads so we can stop merging PR for a little bit to give you time to rebase

**Comment by MosCD3** [comment]
rebase done

**Review by bryce-mcmath — APPROVED** [review]

**Pull request merged**